invertedai/large/initialize.py

```diff
@@ -274,29 +240,147 @@ def get_number_of_agents_per_region_by_drivable_area(
+    center it is closest. Agents will be placed "into" the region that is closest even if 
+    it is not within the bounds of the region. The length of the agent_states list must be
+    equal or less than the length of agent_properties. To remain compliant with :func:`initialize`, 
+    agents with defined agent states are placed at the beginning of the list. Optionally 
+    using the return_region_index parameter will return a list indicating in which region 
+    the agent is placed to preserve agent indexing. A random seed parameter is included for 
+    repeatability.
+
+    Arguments
+    ----------
+    regions:
+        A list of Regions with bounds and centre defined for which agents are associated. 
+
+    agent_states:
+        Please refer to the documentation of :func:`drive` for information on this parameter.
+
+    agent_properties:
+        Please refer to the documentation of :func:`drive` for information on this parameter.
+
+    return_region_index:
+        Whether to map the region in which agents of the same index have been placed. Returns 
+        a list of the same size as the agent_properties parameter.
+    """
+    num_agent_states = len(agent_states)
+    num_regions = len(regions)
+    assert num_regions > 0, "Invalid parameter: number of regions must be greater than zero."
+    assert len(agent_properties) >= num_agent_states, "Invalid parameters: number of agent properties must be larger than number agent states."
+
+    if return_region_index: 
+        region_map = []
+    else:
+        region_map = None
+
+    if len(agent_states) > 0: 
+        region_agent_states_lengths = [len(region.agent_states) for region in regions]
+
+        for i, (prop, state) in enumerate(zip(agent_properties[:num_agent_states],agent_states)):
+            region_distances = []
+            for region in regions:
+                region_distances.append(sqrt((state.center.x-region.center.x)**2 + (state.center.y-region.center.y)**2))
+
+            closest_region_index = region_distances.index(min(region_distances))
+            insert_index = region_agent_states_lengths[closest_region_index]
+            region_agent_states_lengths[closest_region_index] += 1
+            regions[closest_region_index].agent_properties.insert(insert_index,prop)
+            regions[closest_region_index].agent_states.insert(insert_index,state)
+
+            if return_region_index: region_map.append(tuple([closest_region_index,insert_index]))
+
+    if random_seed is not None: seed(random_seed)
+    for prop in agent_properties[num_agent_states:]:
+        random_region_index = randint(0,num_regions-1)
+        regions[random_region_index].agent_properties.append(prop)
+
+        if return_region_index: region_map.append(tuple([random_region_index,len(regions[random_region_index].agent_properties)-1]))
+
+    return regions, region_map
+
+def _consolidate_all_responses(
+    all_responses: Optional[InitializeResponse] = None,
```

Comment:
I see all_responses is assumed to be a list below with `len` and indexing, but it's of type `InitializeResponse`, am I tripping here?

Reply:
Huh that is weird that it didn't cause an issue during testing...
Anyways it is fixed now